django/contrib/postgres/aggregates/general.py

```diff
@@ -41,28 +68,36 @@ class BoolOr(Aggregate):
     output_field = BooleanField()
 
 
-class JSONBAgg(OrderableAggMixin, Aggregate):
+class JSONBAgg(DeprecatedConvertValueMixin, OrderableAggMixin, Aggregate):
     function = 'JSONB_AGG'
     template = '%(function)s(%(distinct)s%(expressions)s %(ordering)s)'
     allow_distinct = True
     output_field = JSONField()
 
-    def convert_value(self, value, expression, connection):
-        if not value:
-            return '[]'
-        return value
+    # RemovedInDjango50Warning
+    deprecation_value = '[]'
+    deprecation_msg = (
+        "In Django 5.0, JSONBAgg() will return None instead of an empty list "
+        "if there are no rows. Pass default=None to opt into the new behavior "
+        "and silence this warning or default=Value('[]') to keep the previous "
```

Comment:
Umm, yes these is otherwise it'll be treated as a column reference?

Reply:
If we want to take that approach we'll need to wrap `default` in `Value()` if it isn't already an expression and force use of `F()` because it could be useful to default to the value of a another column?

But that feels inflexible - note how `StringAgg.__init__()` handles `delimiter` which prevents use of some other expression:

https://github.com/django/django/blob/59942a66ceb79868cb91844df3a72a24c63e39fa/django/contrib/postgres/aggregates/general.py#L61-L63

Reply:
Good catch, but the same issue is with `''`. I added tests for recommended defaults and fix docs.